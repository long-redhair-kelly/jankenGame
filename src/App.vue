<script lang="ts">
export default {
  data() {
    return {
      handArray: ["ぐー", "ちょき", "ぱー"],
      myHand: "",
      enemyHand: "",
      result: "",
      resultColor: "",
      win: 0,
      lose: 0,
      draw: 0,
      turn: 0,
    };
  },
  methods: {
    judge(myHand: string) {
      const number = Math.floor(Math.random() * this.handArray.length);
      this.enemyHand = this.handArray[number];
      this.myHand = myHand;

      // 勝ちのパターン
      if (
        (myHand === "ぐー" && this.enemyHand === "ちょき") ||
        (myHand === "ちょき" && this.enemyHand === "ぱー") ||
        (myHand === "ぱー" && this.enemyHand === "ぐー")
      ) {
        this.win++;
        this.result = "You Win!!";
        this.resultColor = "win";
      } else if (
        (myHand === "ぐー" && this.enemyHand === "ぱー") ||
        (myHand === "ちょき" && this.enemyHand === "ぐー") ||
        (myHand === "ぱー" && this.enemyHand === "ちょき")
      ) {
        this.lose++;
        this.result = "You lose...";
        this.resultColor = "lose";
      } else {
        this.draw++;
        this.result = "You draw.";
        this.resultColor = "draw";
      }
      this.turn++;
    },
  },
  computed: {
    colorChange() {
      return this.resultColor;
    },
  },
};
</script>

<template>
  <div class="wrap">
    <h2>じゃんけん</h2>
    <p>{{ turn }}回目のじゃんけん</p>
    <p>あなたの手を選んでください</p>
    <div class="judgeWrapper">
      <button @click="judge('ぐー')">ぐー</button>
      <button @click="judge('ちょき')">ちょき</button>
      <button @click="judge('ぱー')">ぱー</button>
    </div>
    <div>
      <p>あなたの手は: {{ myHand }} でした</p>
      <p>相手の手は: {{ enemyHand }} でした</p>
    </div>
    <div>
      <p>
        結果: <span :class="colorChange">{{ result }}</span>
      </p>
      <p>勝ち数: {{ win }}</p>
      <p>負け数: {{ lose }}</p>
      <p>引き分け数: {{ draw }}</p>
    </div>
  </div>
</template>
