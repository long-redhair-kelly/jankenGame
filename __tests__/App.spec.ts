import { render } from "@testing-library/vue";
import App from "../src/App";

describe("App.vue", () => {
  it("App スナップショット", () => {
    const wrapper = render(App);
    expect(wrapper).toMatchSnapshot();
  });
});
