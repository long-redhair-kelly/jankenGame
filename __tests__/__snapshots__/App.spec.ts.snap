// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue App スナップショット 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="wrap"
      >
        <h2>
          じゃんけん
        </h2>
        <p>
          0回目のじゃんけん
        </p>
        <p>
          あなたの手を選んでください
        </p>
        <div
          class="judgeWrapper"
        >
          <button>
            ぐー
          </button>
          <button>
            ちょき
          </button>
          <button>
            ぱー
          </button>
        </div>
        <div>
          <p>
            あなたの手は:  でした
          </p>
          <p>
            相手の手は:  でした
          </p>
        </div>
        <div>
          <p>
             結果: 
            <span
              class=""
            />
          </p>
          <p>
            勝ち数: 0
          </p>
          <p>
            負け数: 0
          </p>
          <p>
            引き分け数: 0
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrap"
    >
      <h2>
        じゃんけん
      </h2>
      <p>
        0回目のじゃんけん
      </p>
      <p>
        あなたの手を選んでください
      </p>
      <div
        class="judgeWrapper"
      >
        <button>
          ぐー
        </button>
        <button>
          ちょき
        </button>
        <button>
          ぱー
        </button>
      </div>
      <div>
        <p>
          あなたの手は:  でした
        </p>
        <p>
          相手の手は:  でした
        </p>
      </div>
      <div>
        <p>
           結果: 
          <span
            class=""
          />
        </p>
        <p>
          勝ち数: 0
        </p>
        <p>
          負け数: 0
        </p>
        <p>
          引き分け数: 0
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
